group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/Assignment.java

```diff
@@ -29,7 +30,7 @@
 /**
  * An immutable assignment for a member.
  */
-public class Assignment {
+public class Assignment implements MemberAssignment {
```

Comment:
could you help me understand the difference between Assignment and MemberAssignmentImpl?

Also, the name seems a bit vague given that it implements MemberAssignment

Reply:
That's a good point. There are similar at the moment. Assignment should get extra fields in the future when we implement the client side assignor (if we ever do it) while MemberAssignment will stay as it is. We could consider merging them in the future.

Reply:
So, Assignment is different from MemberAssignmentImpl in that it will also be used for client side assignors, possibly in the future, whereas MemberAssignmentImpl is only used for server side assignors. is this correct?

I think it might be worth adding a short comment or perhaps merging it now in a follow up PR.